### PR TITLE
Ucf update rebase

### LINKED ISSUE #730: Baseline UCF101 model uses float64 for clip_values; triggers bug in ART for some attacks
The baseline UCF101 model has clip values that are float64. This is normally fine, except setting `.attack.kwargs.num_random_init = 1` in `ucf101_baseline_pretrained.json` triggers a bug in ART.  I believe is a bug in ART so I filed an issue there, https://github.com/Trusted-AI/adversarial-robustness-toolbox/issues/582, but I think this is a (very, very tiny) memory issue in Armory and may have some caused others some issues with using stronger attacks on the baseline model.

### LINKED ISSUE #776: Verify benign accuracy of UCF101 baseline model with canonical preprocessing
Ensure that [new refactor of preprocessing](https://github.com/twosixlabs/armory/pull/734) didn't degrade accuracy on clean samples.

### PR DESCRIPTION
**Pull request by davidslater**
Fixes #730
Fixes #776

### CODE REVIEW

File: armory/data/datasets.py [hunk 1]
```diff
@@ -680,12 +680,73 @@ def resisc45(
...
+        if a != t and t is not None:
+            raise ValueError(f"shape {actual} does not match shape {target}")
+
+
+class UCF101Context:
+    def __init__(self):
+        self.nb_classes = 101
+        self.input_type = np.uint8
+        self.x_shape = (None, 240, 320, 3)
+        self.frame_rate = 25
```
Comment (lcadalzo) [inline armory/data/datasets.py:708]: It doesn't appear that this variable is used anywhere. Was it intended to be used in some sort of check, or just there for informative purposes?

The same applies to `self.nb_classes`.

Comment (davidslater) [reply armory/data/datasets.py:708]: Primarily informational. Some models could use frame rate. I figure that this sort of object would make the most sense to store this sort of info.

**Comment by lcadalzo** [comment]
Has #776 been confirmed?

**Comment by davidslater** [comment]
No, I was just linking it as a requirement for merging.

**Comment by lcadalzo** [comment]
The results I'm seeing do confirm #776. 

250 samples of `ucf101_baseline_pretrained.json` on `ucf-update-rebase` branch: 
`Average categorical_accuracy on benign test examples: 94.40%`
`Average top_5_categorical_accuracy on benign test examples: 98.80%`

And 250 samples of `ucf101_baseline_pretrained.json` on master branch (1cd0f1e):
`Average categorical_accuracy on benign test examples: 95.42%`
`Average top_5_categorical_accuracy on benign test examples: 99.17%`

**Comment by davidslater** [comment]
Great!

**Comment by lcadalzo** [comment]
Unrelated to canonical preprocessing changes, but for [this line](https://github.com/davidslater/armory/blob/ucf-update-rebase/armory/baseline_models/pytorch/ucf101_mars.py#L223), in the case where `trained` is True and `weights_path` is None, is it supposed to carry on without loading any checkpoint, or perhaps raise an exception? Unless there are some default weights loaded during the call to `generate_model(opt)`

**Comment by davidslater** [comment]
It's a bit of a misnomer, I think. That case is so that you can generate a randomly initialized model, if I recall correctly.

**Review by lcadalzo — APPROVED** [review]
LGTM

**Pull request merged**